site/src/docs/python/basic-usage.md

```diff
@@ -0,0 +1,246 @@
+
+```py
+from dataclasses import dataclass
+from typing import Literal
+
+@dataclass
+class Sentiment:
+    """
+    The following is a schema definition for determining the sentiment of a some user input.
+    """
+
+    sentiment: Literal["negative", "neutral", "positive"]
+```
+
+Here, we're saying that the `sentiment` attribute has to be one of three possible strings: `negative`, `neutral`, or `positive`.
+We did this with [the `typing.Literal` hint](https://docs.python.org/3/library/typing.html#typing.Literal).
+
+We defined `Sentiment` as a `@dataclass` so we could have all of the conveniences of standard Python objects - for example, to access the `sentiment` attribute, we can just write `value.sentiment`.
+If we declared `Sentiment` as a `TypedDict`, TypeChat would provide us with a `dict`.
+That would mean that to access the value of `sentiment`, we would have to write `value["sentiment"]`.
+
+Note that while we used [the built-in `typing` module](https://docs.python.org/3/library/typing.html), [`typing_extensions`](https://pypi.org/project/typing-extensions/) is supported as well.
+TypeChat also understands constructs like `Annotated` and `Doc` to add comments to individual attributes.
+
+## Creating a Validator
+
+A validator really has two jobs generating a textual schema for language models, and making sure any data fits a given shape.
+The built-in validator looks roughly like this:
+
+```py
+class TypeChatValidator(Generic[T]):
+    """
+    Validates an object against a given Python type.
+    """
+
+    def __init__(self, py_type: type[T]):
+        """
+        Args:
+
+            py_type: The schema type to validate against.
+        """
+        ...
+
+    def validate_object(self, obj: object) -> Result[T]:
+        """
+        Validates the given Python object according to the associated schema type.
+
+        Returns a `Success[T]` object containing the object if validation was successful.
+        Otherwise, returns a `Failure` object with a `message` property describing the error.
+        """
+        ...
+```
+
+To construct a validator, we just have to pass in the type we defined:
+
+```py
+import schema as sentiment
+validator = TypeChatValidator(sentiment.Sentiment)
+```
+
+## Creating a JSON Translator
+
+A `TypeChatJsonTranslator` brings all these concepts together.
+A translator takes a language model, a validator, and our expected type, and provides a way to translate some user input into objects following our schema.
+To do so, it crafts a prompt based on the schema, reaches out to the model, parses out JSON data, and attempts validation.
+Optionally, it will craft repair prompts and retry if validation fails.
+
+```py
+translator = TypeChatJsonTranslator(model, validator, sentiment.Sentiment)
+```
+
+When we are ready to translate a user request, we can call the `translate` method.
+
+```ts
+translator.translate("Hello world! 🙂");
+```
+
+We'll come back to this.
+
+## Creating the Prompt

```

Comment:
"Prompt" is an ambiguous term here. Does it refer to the user input prompt (only used when file_path is None) or the prompt for the LLM?